src/west/manifest.py

```diff
@@ -482,9 +491,24 @@ def _load_self(self, path_hint):
             # restriction in the future.
             assert self.topdir
 
-        return ManifestProject(path=slf.get('path', path_hint),
-                               topdir=self.topdir,
-                               west_commands=slf.get('west-commands'))
+        # Sanity-check /self/import. We don't do anything else at this
+        # stage: the imports themselves are done last, after
+        # /projects/X/imports and self._data['projects'] are both done.
+        slf_import = slf.get('import')
+        if not (slf_import is None or isinstance(slf_import, bool)):
```

Comment:
Not clear to us what this check is for

Reply:
This code needs to be reworked. I was originally only going to support this feature if there was a real installation on disk, but then I realized that importers are general enough to not require that.

The purpose of this check is to say that if you're doing an import that has some effect (unlike `import: false` or `self: import: true`, which are ignored / have no effect), then you must have also provided the topdir of the installation.

But we should just call into the importer here instead.